extensions.html

```diff
@@ -182,19 +184,146 @@ <h2>
         <dfn>GamepadHapticActuator</dfn> Interface
       </h2>
       <p>
-        Each <code>GamepadHapticActuator</code> corresponds to a motor or other
-        actuator that can apply a force for the purposes of haptic feedback.
+        A {{GamepadHapticActuator}} corresponds to a configuration of motors or
+        other actuators that can apply a force for the purposes of haptic
+        feedback.
       </p>
       <pre class="idl">
         [Exposed=Window]
         interface GamepadHapticActuator {
           readonly attribute GamepadHapticActuatorType type;
+          boolean canPlayEffectType(GamepadHapticEffectType type);
+          Promise&lt;GamepadHapticsResult&gt; playEffect(
+              GamepadHapticEffectType type,
+              optional GamepadEffectParameters params = {});
           Promise&lt;boolean&gt; pulse(double value, double duration);
```

Comment:
isn't "pulse" also an effect? Maybe it could be `playEffect("pulse")`... otherwise, you end up with situations like:

```
.pulse(a,b);
.playEffect(obj);
```

Which would conflict with each other. Having a single `.playEffect()` call would avoid that. 

```
.playEffect("dual-rumble").then(()=> .playEffect("pulse"))
```

Reply:
I'm fine with removing pulse() since I think there are no actual implementations. Chromium doesn't implement it, Firefox has IDL but it's not hooked up on any platforms.

I'm not sure we even need a "pulse" effect. In practice, far more gamepads use two-channel "dual-rumble" haptics than single-channel "pulse" haptics. Reading the [Firefox bug](https://bugzilla.mozilla.org/show_bug.cgi?id=680289) it's not clear to me how we arrived at the single-channel version currently in extensions.html.

I'd lean toward removing pulse() and having applications send "dual-rumble" effects with equal strongMagnitude and weakMagnitude values since that's what the user agent would likely do anyway.

Reply:
Agree. It would be nice to have a single way to activate the actuator.

Reply:
The quest browser implemented pulse and it's widely use by WebXR experiences so it ca't be removed